wps_shared/wps_shared/db/crud/snow.py

```diff
@@ -27,19 +27,17 @@ async def get_last_processed_snow_by_processed_date(session: AsyncSession, proce
     result = await session.execute(stmt)
     return result.first()
 
-async def get_most_recent_processed_snow_by_date(session: AsyncSession, target_date: datetime, snow_source: SnowSourceEnum = SnowSourceEnum.viirs) -> ProcessedSnow:
+
+async def get_most_recent_processed_snow_by_date(session: AsyncSession, target_date: date | datetime, snow_source: SnowSourceEnum = SnowSourceEnum.viirs) -> ProcessedSnow:
     """Retrieve the most recent record prior or equal to the provided date.
 
     :param target_date: The date of interest
-    :type target_date: datetime
+    :type target_date: date or datetime
     :param snow_source: The source of snow data of interest.
     :type snow_source: SnowSourceEnum
     :return: A record containing the last date for which snow data from the specified source was successfully processed.
     :rtype: ProcessedSnow
     """
-    stmt = select(ProcessedSnow)\
-    .where(ProcessedSnow.snow_source == snow_source)\
-    .where(ProcessedSnow.for_date <= target_date)\
-    .order_by(ProcessedSnow.for_date.desc())
+    stmt = select(ProcessedSnow).where(ProcessedSnow.snow_source == snow_source).where(ProcessedSnow.for_date <= target_date).order_by(ProcessedSnow.for_date.desc())
```

Comment:
Are there any situations (aka times of day) where comparison of the timezone aware datetime from the database against a vanilla `date` object could lead to a problem? Or are we safe because of the times of day that we receive and process hfi rasters?

Reply:
I'm actually a little surprised SQLAlchemy or Postgres allows this comparson!

Reply:
Ignore this, I see Conor beat me to it!

Reply:
You bring up a good point, I'd prefer if we just chose `date`, but I also don't have a counter example to provide for motivation.